Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageContainerView.swift

```diff
@@ -271,43 +270,42 @@ public struct MessageContainerView<Factory: ViewFactory>: View {
                         : nil
                 )
 
-                if !message.isRightAligned {
+                if !messageViewModel.isRightAligned {
                     MessageSpacer(spacerWidth: spacerWidth)
                 }
             }
         }
         .padding(
             .top,
-            topReactionsShown && !isMessagePinned ? messageListConfig.messageDisplayOptions.reactionsTopPadding(message) : 0
+            topReactionsShown && !messageViewModel.isPinned ? messageListConfig.messageDisplayOptions
+                .reactionsTopPadding(message) : 0
         )
         .padding(.horizontal, messageListConfig.messagePaddings.horizontal)
-        .padding(.bottom, showsAllInfo || isMessagePinned ? paddingValue : groupMessageInterItemSpacing)
+        .padding(.bottom, showsAllInfo || messageViewModel.isPinned ? paddingValue : groupMessageInterItemSpacing)
         .padding(.top, isLast ? paddingValue : 0)
-        .background(isMessagePinned ? Color(colors.pinnedBackground) : nil)
-        .padding(.bottom, isMessagePinned ? paddingValue / 2 : 0)
+        .background(messageViewModel.isPinned ? Color(colors.pinnedBackground) : nil)
+        .padding(.bottom, messageViewModel.isPinned ? paddingValue / 2 : 0)
         .transition(
             message.isSentByCurrentUser ?
                 messageListConfig.messageDisplayOptions.currentUserMessageTransition :
                 messageListConfig.messageDisplayOptions.otherUserMessageTransition
         )
         .accessibilityElement(children: .contain)
         .accessibilityIdentifier("MessageContainerView")
+        .id(message.id + messageViewModel.textContent)
+        .environment(\.messageViewModel, messageViewModel)
```

Comment:
this is needed only for the link detector text view, right? Maybe we add V5 todo?

Reply:
I also thought about this, but doing this now, it will be some significant changes, especially because we need to pass the view model furtherdown at least unil `MessageView`. So for now lets keep it like this to minimize the changes 👍